src/networkrequest.cpp

```diff
@@ -133,6 +133,41 @@ NetworkRequest* NetworkRequest::createForAuthenticationVerification(
   return r;
 }
 
+// static
+NetworkRequest* NetworkRequest::createForAdjustProxy(
+    QObject* parent, const QString& method, const QString& route,
+    const QList<QPair<QString, QString>>& headers, const QString& parameters) {
+  Q_ASSERT(parent);
+
+  NetworkRequest* r = new NetworkRequest(parent, 200, false);
+
+  r->m_request.setHeader(QNetworkRequest::ContentTypeHeader,
+                         "application/json");
+
+  QUrl url(apiBaseUrl());
+  url.setPath("/api/v1/vpn/adjust");
+  r->m_request.setUrl(url);
+
+  QJsonObject headersObj;
+  for (QPair<QString, QString> header : headers) {
+    if (header.first != "Host" && header.first != "Content-Length") {
```

Comment:
Compare the headers using a case-insensitive algorithm:
```
  if (QString::compare(header.first, "host", Qt::CaseInsensitive) != 0 && QString::compare(header.first, "content-length", Qt::CaseInsensitive)) {
  ..
}
```

But probably, the best approach is to skip these headers when parsing them,